core/src/main/scala/com/databricks/labs/remorph/coverage/encoders.scala

```diff
@@ -0,0 +1,76 @@
+package com.databricks.labs.remorph.coverage
+
+import com.databricks.labs.remorph.coverage.estimation.{Rule => EstRule, _}
+import com.databricks.labs.remorph.discovery.{Fingerprint, QueryType, WorkloadType}
+import com.databricks.labs.remorph.intermediate._
+import io.circe._
+import io.circe.generic.extras.semiauto._
+import io.circe.generic.extras.Configuration
+import io.circe.syntax._
+
+import java.io.{PrintWriter, StringWriter}
+import java.sql.Timestamp
+import java.time.Duration
+
+trait ErrorEncoders {
+
+  implicit val codecConfiguration: Configuration =
+    Configuration.default.withDiscriminator("$type").withSnakeCaseMemberNames
```

Comment:
correct me if i'm wrong, but all of this is required because we have `RemorphError` trait that we serialize?

better option is:

- not to include `RemorphError` in json serialization
- change the interface of it to always convert into `problems: Seq[Problem]`, even if it's a single error.
- it's fine to have `RemorphError` trait, but we always have to post-process it into `problems: Seq[Problem]` before we do serialization.
- we have less risks of nested structs leaking into JSON repr and we're closer to tabular form of this
- `Problem.code` will essentially be a simple name of class that you're putting into `$type` discriminator
- we may not need `circle` library then :) 

```
case class Problem(code: String, message: String)
sealed trait RemorphError {
  def problems: Seq[Problem]
}
```